### PR TITLE
DEV-4848: Implement Phone Number input

### PR DESCRIPTION
**Pull request by GuiMend**
Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)
n/a
#### What's this PR do?
- Implements Phone Number input and add it to Contributor Portal Edit page
#### Why are we doing this? How does it help us?
- Improve the UX of adding a phone number
#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?
- Yes
#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
- yes
#### Have automated unit tests been added? If not, why?
- yes
#### How should this change be communicated to end users?
- n/a
#### Are there any smells or added technical debt to note?
- no
#### Has this been documented? If so, where?
- no
#### What are the relevant tickets? Add a link to any relevant ones.
- https://news-revenue-hub.atlassian.net/browse/DEV-4848
#### Do any changes need to be made before deployment to production (adding environment variables, for example)? If so, open a ticket and link it to the ticket for this PR and list it here:
- no
#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:
- no

### CODE REVIEW

**Review by nrh-cklimas — CHANGES_REQUESTED** [review]

File: spa/src/components/base/TextField/PhoneTextField.test.tsx [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import { axe } from 'jest-axe';
+import { render, screen } from 'test-utils';
+import PhoneTextField, { PhoneTextFieldProps } from './PhoneTextField';
+
+function tree(props?: Partial<PhoneTextFieldProps>) {
+  return render(<PhoneTextField id="mock-label" label="mock-label" {...props} />);
+}
+
+describe('PhoneTextField', () => {
```
Comment (nrh-cklimas) [inline spa/src/components/base/TextField/PhoneTextField.test.tsx:9]: Test coverage says we should add tests around choosing a country from the flag menu.

File: spa/src/components/base/TextField/PhoneTextField.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
+import countryCodes from 'country-code-lookup';
+import { Select } from '../Select';
+import { TextField, TextFieldProps } from './TextField';
+import { CountryIso2, FlagImage, usePhoneInput } from 'react-international-phone';
+import styled from 'styled-components';
+import { InputAdornment } from '@material-ui/core';
+
+export type PhoneTextFieldProps = Omit<TextFieldProps, 'onChange'> & {
+  onChange?: (...event: any[]) => void;
```
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/base/TextField/PhoneTextField.tsx:9]: I'm not sure I understand how this component would be used in 2 different forms. The prop onChange is only being used inside `usePhoneInput`
Btw, the TS for this `onChange` I got from `react-hook-form`.

File: spa/src/components/base/TextField/PhoneTextField.test.tsx (hunk 1 above)
Comment (GuiMend) [reply spa/src/components/base/TextField/PhoneTextField.test.tsx:10]: [e1b117d](https://github.com/newsrevenuehub/rev-engine/pull/1546/commits/e1b117d38f11467e5dd68f5e6434fc066cc3c0a9)

File: spa/src/components/base/TextField/PhoneTextField.test.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    ['US number', '+16195357597', '+1 (619) 535-7597'],
+    ['Brazil number', '+5548988009988', '+55 (48) 988009988']
+  ])('%s: renders existing formatted phone number', async (_, number, formatted) => {
+    const onChange = jest.fn();
+    tree({ value: number, onChange });
+    expect(screen.getByRole('textbox')).toHaveValue(formatted);
+    expect(onChange).not.toHaveBeenCalled();
+  });
+
+  it('calls onChange with default number', () => {
```
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/base/TextField/PhoneTextField.test.tsx:26]: [e1b117d](https://github.com/newsrevenuehub/rev-engine/pull/1546/commits/e1b117d38f11467e5dd68f5e6434fc066cc3c0a9)

File: spa/src/components/base/TextField/PhoneTextField.tsx [hunk 4]
```diff
@@ -32,7 +32,6 @@ const StyledSelect = styled(Select)`
 export function PhoneTextField({ onChange, value, ...props }: PhoneTextFieldProps) {
   const { inputValue, handlePhoneValueChange, inputRef, country, setCountry } = usePhoneInput({
     defaultCountry: 'us',
-    forceDialCode: true,
```
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/base/TextField/PhoneTextField.tsx]: [e1b117d](https://github.com/newsrevenuehub/rev-engine/pull/1546/commits/e1b117d38f11467e5dd68f5e6434fc066cc3c0a9)

File: spa/src/components/content/contributor-portal/ContributorPortal.tsx [hunk 5]
```diff
@@ -53,6 +53,11 @@ const ContributorPortal = ({ revenueProgram }: ContributorPortalProps) => {
     [contact_email, contact_phone, revenueProgram?.contact_email, revenueProgram?.contact_phone]
   );
 
+  const onlyCountryCode = contact_phone.length > 0 && contact_phone.length < 6;
```
Comment (nrh-cklimas) [inline spa/src/components/content/contributor-portal/ContributorPortal.tsx:56]:
```suggestion
  const onlyCountryCode = /^\+\d+$/.test(contact_phone);
```

I might have this wrong, but I think this approach is a little more flexible.

File: spa/src/components/base/TextField/PhoneTextField.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/base/TextField/PhoneTextField.tsx:9]: I think, but I'm not 100% sure, that there are two onChange possibilities here and they receive different arguments.

- onChange on the TextField component sends a React.ChangeEvent.
- onChange coming from usePhoneInput sends just the value.

There's [a MUI example](https://react-international-phone.vercel.app/docs/Advanced%20Usage/useWithUiLibs#material-ui-example) in react-international-phone's docs that shows this, though in this example it only has a single onChange prop, the one provided by the hook. It's possible we only need to use the onChange from the hook to capture all changes. I'd try it out and see.

File: spa/src/components/content/contributor-portal/ContributorPortal.tsx (hunk 5 above)
Comment (GuiMend) [reply spa/src/components/content/contributor-portal/ContributorPortal.tsx:56]: This doesn't work @nrh-cklimas. `usePhoneInput` has 2 phone strings, `{ phone: string; inputValue: string }`
1. inputValue is the formatted one, +1 (555) 555555 (this is the one displayed in the input)
2. phone is unformatted, +1555555555 (this one is the actual value in `contact_phone`)

Your suggestion works with `inputValue`, but not with `phone` (which is what we use).

File: spa/src/components/base/TextField/PhoneTextField.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/base/TextField/PhoneTextField.tsx:9]: > onChange on the TextField component sends a React.ChangeEvent.

Yes, but this onChange is being used with `onChange={handlePhoneValueChange}` (we do not override this with the onChange prop)

> onChange coming from usePhoneInput sends just the value

The component prop, `onChange`, is used inside the `usePhoneInput` hook

> There's [a MUI example](https://react-international-phone.vercel.app/docs/Advanced%20Usage/useWithUiLibs#material-ui-example) in react-international-phone's docs that shows this, though in this example it only has a single onChange prop, the one provided by the hook. It's possible we only need to use the onChange from the hook to capture all changes. I'd try it out and see.

We are using in the same way as the example. Are you mistaken about something or am I not getting what you mean?

[the remark this answers is not in the document]
Comment (nrh-cklimas) [reply spa/src/components/base/TextField/PhoneTextField.tsx:9]: Ah, I get it now. I think the typing on `onChange` should be `(value: string) => void` in this case.

File: spa/src/components/content/contributor-portal/ContributorPortal.tsx (hunk 5 above)
Comment (nrh-cklimas) [reply spa/src/components/content/contributor-portal/ContributorPortal.tsx:56]: As discussed on Slack, I was seeing weird behavior with leftover data from the previous PR. I don't think that's going to be a problem here so resolving.

File: spa/src/components/base/TextField/PhoneTextField.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuiMend) [reply spa/src/components/base/TextField/PhoneTextField.tsx]: [89cd623](https://github.com/newsrevenuehub/rev-engine/pull/1546/commits/89cd623913a2e8ff0300ad6ddbc6a026089d9f13)

**Review by nrh-cklimas — APPROVED** [review]

**Pull request merged**